INTMDB-435: Add ForceNew to audit if the project id changes

martinstibbe opened the pull request:
## Description

INTMDB-435: Add ForceNew to audit if the project id changes

Link to any related issue(s):
https://github.com/mongodb/terraform-provider-mongodbatlas/issues/759

[INTMDB-435](https://jira.mongodb.org/browse/INTMDB-435)

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments

martinstibbe committed:
Add ForceNew on project_id

andreaangiolillo commented on mongodbatlas/resource_mongodbatlas_auditing.go:
```diff
@@ -30,6 +30,7 @@ func resourceMongoDBAtlasAuditing() *schema.Resource {
 			"project_id": {
 				Type:     schema.TypeString,
 				Required: true,
+				ForceNew: true,
```
Could you update the documentation to point out this new behaviour?

martinstibbe replied:
Added Note to parameter

Zuhairahmed reviewed (changes requested):
Agree with @andreaangiolillo, can you update: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/auditing

Also can you confirm that no changes need to be made to resource_mongodbatlas_auditing_test.go?

Lastly, minor nit, I added link to INTMDB ticket in Description section of PR if we can continue to do that going forward as well. Thanks!

martinstibbe committed:
Externalize Delete to reuse for project creation failures

martinstibbe committed:
Revert "Externalize Delete to reuse for project creation failures"

This reverts commit fb11204e56bbd2e7e85e8543bc57d5aba71d70c2.

martinstibbe committed:
Add note to docs about ForceNew

martinstibbe commented:
> Agree with @andreaangiolillo, can you update: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/auditing
> 
> Also can you confirm that no changes need to be made to resource_mongodbatlas_auditing_test.go?
> 
> Lastly, minor nit, I added link to INTMDB ticket in Description section of PR if we can continue to do that going forward as well. Thanks!

Added Ticket # to Description
No changes needed for resource_mongodbatlas_auditing_test

Zuhairahmed commented on website/docs/r/auditing.html.markdown:
```diff
@@ -23,7 +23,7 @@ resource "mongodbatlas_auditing" "test" {
 ## Argument Reference
 
-* `project_id` - (Required) The unique ID for the project to configure auditing.
+* `project_id` - (Required) The unique ID for the project to configure auditing. **Note! When changing this value to a different project_id it will delete the current audit settings for the original project that was assigned to.**
```
```suggestion
* `project_id` - (Required) The unique ID for the project to configure auditing. **Note: When changing this value to a different project_id it will delete the current audit settings for the original project that was assigned to.**
```

Zuhairahmed reviewed (approved):
Minor doc update, not blocking. LGTM!

martinstibbe committed:
Update website/docs/r/auditing.html.markdown

Co-authored-by: Zuhair Ahmed <zuhair.ahmed@mongodb.com>

martinstibbe committed:
Merge branch 'master' into INTMDB-435
